### ISSUE TITLE
Update reference for curvature

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/xdem#577]
At [this line](https://github.com/GlacioHack/xdem/blob/02c20268ecfececc2be75e07597ec66997902743/doc/source/terrain.md?plain=1#L92), the doc says that the reference for the curvature is [Zevenbergen and Thorne (1987)](http://dx.doi.org/10.1002/esp.3290120107). However, equation 16 in this ref is different from the implementation in xdem at [this line](https://github.com/GlacioHack/xdem/blob/02c20268ecfececc2be75e07597ec66997902743/xdem/terrain.py#L997) (-2(D + E)). This equation is instead used in [Moore et al. (1991)](https://doi-org.sid2nomade-2.grenet.fr/10.1002/hyp.3360050103) (equation 16).

Also I noticed a mistake in the description of the curvature. "convex (like a mountain peak)" and "concave (like a through or a valley bottom)" are wrong and should be "concave (like a mountain peak)" and "convex (like a through or a valley bottom)". But one needs to check which one corresponds to negative/positive curvature.

**Comment by rhugonnet** [comment]
Good catch. This means RichDEM is also wrong on that formulation, as we test against it and it cites Z&T:
https://github.com/GlacioHack/xdem/blob/34e5049ae76a858b9c95c8763d9763b298c56e3c/tests/test_terrain.py#L208.

Should we then implement both the formulations of Zevenberg and Thorne and of Moore et al.?  It would be weird to have all curvatures based on Z&T except the one.

**Comment by adehecq** [comment]
> Should we then implement both the formulations of Zevenberg and Thorne and of Moore et al.? It would be weird to have all curvatures based on Z&T except the one.

It would make sense indeed!

**Comment by rhugonnet** [comment]
After more reading, Moore et al. (1991) introduces this specific term, but it is based on the 9-term 3x3 window fitting method of Zevenberg and Thorne (1987), which doesn't introduce anything else than a directional curvature function.
So we should simply cite both at once. Adding this in #486.

**Comment by rhugonnet** [comment]
Finally, convex in 3D seems to correspond more to a bulging landform which would contain points, and concave to a hole that wouldn't (see figures on Google), so I think the original description is good.
